source/payload-interfaces/index.rst

```diff
@@ -360,6 +360,158 @@ Set to 0 to use the default baud rate 115200.
+
+  typedef struct {
+    PLD_GENERIC_HEADER   PldHeader;
+    UINT8                Count;
+    PLD_PCI_ROOT_BRIDGE  RootBridge[0];
+  } PLD_PCI_ROOT_BRIDGES;
+
+  typedef struct {
+    UINT32                       Segment;
+    UINT64                       Supports;
+    UINT64                       Attributes;
+    BOOLEAN                      DmaAbove4G;
+    BOOLEAN                      NoExtendedConfigSpace;
+    BOOLEAN                      ResourceAssigned;
+    UINT64                       AllocationAttributes;
+    PLD_PCI_ROOT_BRIDGE_APERTURE Bus;
+    PLD_PCI_ROOT_BRIDGE_APERTURE Io;
+    PLD_PCI_ROOT_BRIDGE_APERTURE Mem;
+    PLD_PCI_ROOT_BRIDGE_APERTURE MemAbove4G;
+    PLD_PCI_ROOT_BRIDGE_APERTURE PMem;
+    PLD_PCI_ROOT_BRIDGE_APERTURE PMemAbove4G;
+    UINT32                       HID;
+    UINT32                       UID;
+  } PLD_PCI_ROOT_BRIDGE;
+
+  //
+  // (Base > Limit) indicates an aperture is not available.
+  //
+  typedef struct {
+    //
+    // Base and Limit are the device address instead of host address when
+    // Translation is not zero
+    //
+    UINT64 Base;
+    UINT64 Limit;
+    //
+    // According to UEFI 2.7, Device Address = Host Address + Translation,
+    // so Translation = Device Address - Host Address.
+    // On platforms where Translation is not zero, the subtraction is probably to
+    // be performed with UINT64 wrap-around semantics, for we may translate an
+    // above-4G host address into a below-4G device address for legacy PCIe device
+    // compatibility.
+    //
+    // NOTE: The alignment of Translation is required to be larger than any BAR
+    // alignment in the same root bridge, so that the same alignment can be
+    // applied to both device address and host address, which simplifies the
+    // situation and makes the current resource allocation code in generic PCI
+    // host bridge driver still work.
+    //
+    UINT64 Translation;
+  } PLD_PCI_ROOT_BRIDGE_APERTURE;
+  #pragma pack()
+
+**Member Description**
+
+``PldHeader``
+
+PldHeader.Revision is 1.
+
+PldHeader.Length is 5 + ``Count`` * sizeof (PLD_PCI_ROOT_BRIDGE).
+
+``Count``
+
+Count of root bridges. Number of elements in ``RootBridge`` array.
+
+``RootBridge[i].Segment``
+
+Segment number of the root bridge.
+
+``RootBridge[i].Supports``
+
+Supported attributes. Refer to EFI_PCI_ATTRIBUTE_xxx used by GetAttributes() and SetAttributes() in EFI_PCI_ROOT_BRIDGE_IO_PROTOCOL
+defined in PI Specification.
+
+``RootBridge[i].Attributes``
+
+Initial attributes. Refer to EFI_PCI_ATTRIBUTE_xxx used by GetAttributes() and SetAttributes() in EFI_PCI_ROOT_BRIDGE_IO_PROTOCOL
+defined in PI Specification.
+
+``RootBridge[i].DmaAbove4G``
+
+Root bridge supports DMA above 4GB memory when it's TRUE.
+
+``RootBridge[i].NoExtendedConfigSpace``
+
+Root bridge supports 256-byte configuration space only when it's TRUE.
+Root bridge supports 4K-byte configuration space when it's FALSE.
+
+``RootBridge[i].ResourceAssigned``

```

Comment:
Do you need this at each root bridge level ?  I had thought  we can only do at platform level.
Either all root bridge resources are allocated or none is allocated, it cannot be partial.

Reply:
so maybe move it up level in PLD_PCI_ROOT_BRIDGES ?

Reply:
sure. Good comment!